doc/api/errors.md

```diff
@@ -2344,6 +2344,40 @@ The `package.json` [`"exports"`][] field does not export the requested subpath.
 Because exports are encapsulated, private internal modules that are not exported
 cannot be imported through the package resolution, unless using an absolute URL.
 
+<a id="ERR_PARSE_ARGS_INVALID_OPTION_VALUE"></a>
+
+### `ERR_PARSE_ARGS_INVALID_OPTION_VALUE`
```

Comment:
@shadowspawn @bakkot, I don't have a strong opinion. Chime in and make an argument either way?

Reply:
That's somewhat misleading, I think. It's not that `--with` itself is ambiguous, it's that the _following_ argument is ambiguous in the context following `--with`. But we could maybe find a wording which works there?

We should also have good messages for `--flag=some-value` (where you pass a value to an option which doesn't take one), `--takes-arg` without an arg (where there is nothing following a flag which takes an argument), and the use of positional arguments when `allowPositionals` is false. `ERR_AMBIGUOUS_ARGUMENT` isn't appropriate for any of those three, to my view, but maybe there would be more appropriate options.

Reply:
Short version: would a single custom error code with a fairly freeform message be preferable?

Long version

First a general comment. An author reason for fine grained errors is so author can take custom action for distinct error conditions. I don't know how much customisation will be built on top of `strict:true` though, we anticipate heavy customisation will take place on top of `strict:false` which mostly does not throw and can use a different channel than the "error code" itself.

We dumped the original custom error code as low-value following feedback, and started looking for existing errors: https://github.com/nodejs/node/pull/42675#discussion_r846693562

When adding the initial strict errors, we looked at `ERR_INVALID_ARG_TYPE` and `ERR_INVALID_ARG_VALUE` but couldn't make them fit. Those particular errors are very focused on author errors in javascript arguments to a function or the like, whereas the `parseArgs` errors are referring to user supplied arguments on the command line.

I missed `ERR_AMBIGUOUS_ARGUMENT` when I added that particular error! But I didn't add a new code for that one, I reused a custom error code already added. So no win changing that alone.

`ERR_PARSE_ARGS_INVALID_OPTION_VALUE` is basically freeform to allow custom messages for specific scenarios without extra error codes for subsets of that error. Which leads me to suggest...

We could funnel all the errors through a single custom error?  This does still allow the author to at least determine the error was due to the user supplied arguments.

Reply:
@shadowspawn are you suggesting that, rather than the three errors that have been introduced, `ERR_PARSE_ARGS_INVALID_OPTION_VALUE` is used in each case?

Reply:
A single error yes. But I would also rename it, say `ERR_PARSE_ARGS_INVALID_ARG`.

Reply:
@ljharb made my general point more strongly:

> The program _as well_ as the human needs to discern, so we absolutely shouldn't just use a single error code.
>
> Error codes should generally map to messages 1:1, and a program should never have to parse a message to figure anything out.

Reply:
@RaisinTen given the strong argument @ljharb is making, are you open to leaving the errors as are? I intend to rebase this PR early this week.

Reply:
@bcoe yes, I'm okay with that. It was a non-blocking comment and can be addressed later on too because the parser is still experimental.